src/HotChocolate/Data/src/Data/Filters/Convention/FilterConvention.cs

```diff
@@ -0,0 +1,108 @@
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Reflection;
+using HotChocolate.Types;
+using HotChocolate.Types.Descriptors;
+using HotChocolate.Types.Descriptors.Definitions;
+using System.Collections.Immutable;
+
+namespace HotChocolate.Data.Filters
+{
+    public class FilterConvention :
+        ConventionBase<FilterConventionDefinition>
+        , IFilterConvention
+    {
+        public IImmutableDictionary<int, OperationConvention> Operations { get; private set; }
+
+        private readonly Action<IFilterConventionDescriptor> _configure;
+
+        protected FilterConvention()
+        {
+            _configure = Configure;
+        }
+
+        public FilterConvention(Action<IFilterConventionDescriptor> configure)
+        {
+            _configure = configure
+                ?? throw new ArgumentNullException(nameof(configure));
```

Comment:
Also here we have now 2 styles, which are we gonna choose?

A)

```csharp
_configure = configure
    ?? throw new ArgumentNullException(nameof(configure));
```

B)

```csharp
_configure = configure ?? 
    throw new ArgumentNullException(nameof(configure));
```

@rstaib @PascalSenn

Reply:
B)

Reply:
B)